nixos/modules/services/desktops/dunst.nix

```diff
@@ -0,0 +1,463 @@
+        '';
+      };
+
+    };
+
+
+    frame = {
+      width = mkOption {
+        type = types.int;
+        default = 3;
+        description = ''
+        '';
+      };
+
+      color = mkOption {
+        type = types.str;
+        default = ''#aaaaaa'';
+        description = ''
+        '';
+      };
+
+    };
+
+    shortcuts = {
+      close = mkOption {
+        type = types.str;
+        default = "ctrl+space";
+        description = ''
+        '';
+      };
+
+      closeAll = mkOption {
+        type = types.str;
+        default = "ctrl+shift+space";
+        description = ''
+        '';
+      };
+
+      history = mkOption {
+        type = types.str;
+        default = "ctrl+Escape";
+        description = ''
+        '';
+      };
+
+      context = mkOption {
+        type = types.str;
+        default = "ctrl+shift+period";
+        description = ''
+        '';
+      };
+
+    };
+
+    urgency = {
+      low = {
+        background = mkOption {
+          type = types.str;
+          default = ''#222222'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#888888'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      normal = {
+        background = mkOption {
+          type = types.str;
+          default = ''#285577'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      critical = {
+        background = mkOption {
+          type = types.str;
+          default = ''#900000'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 0;
+          description = ''
+          '';
+        };
+
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.string;
+      default = "";
+      description = "Extra configuration to append to the config file.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.dunst;
+      defaultText = "pkgs.dunst";
+      description = "dunst derivation to use.";
+    };
+  };
+
+
+###### implementation
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.dunst ];
+    systemd.user.services.dunst = {
+      description = "Dunst: lightweight and customizable notification daemon";
+      wantedBy = [ "default.target" ];
```

Comment:
I believe `default.target` is used for user services because there is no standard `graphical.target`, unlike for system services. Havent' checked if we provide such a thing, though.